### PR TITLE
[7/N] MoE Refactor: the implementation of new framework

### PR DESCRIPTION
**Pull request by ch-wan**
## Motivation

This PR implements a new MoE framework introduced in #8715 to streamline the integration of new all-to-all backends or grouped-gemm backends. 

## Modifications

One MoE forward function can be decoupled into three parts: dispatch -> pre-permute -> core runner -> post-permute -> combine. With this PR, a developer can implement dispatch/combine in the FusedMoE module and define their customized moe runner. The detailed workflow is illustrated in #8715.

We offer two modes for implementing layer permutation:

- Mode 1: `register_fused_func` for static fused_op for each pair of a2a x runner backend. This mode is compatible with torch.compile.
- Mode 2: `register_pre_permute` and `register_post_permute` for static runner + dynamic permutation functions. This mode ensures better extensibility.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/layers/quantization/w8a8_fp8.py [hunk 1]
```diff
@@ -266,25 +280,26 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
             layer.w2_weight_scale.data, requires_grad=False
         )
 
+    def create_moe_runner(
+        self, layer: torch.nn.Module, moe_runner_config: MoeRunnerConfig
+    ):
+        self.moe_runner_config = moe_runner_config
+        self.runner = MoeRunner(MoeRunnerBackend.TRITON, moe_runner_config)
```
Comment (BowenBao) [inline python/sglang/srt/layers/quantization/w8a8_fp8.py:287]: It seems `MoeRunner` and `MoeRunnerBackend` and used without import.

Comment (ch-wan) [reply python/sglang/srt/layers/quantization/w8a8_fp8.py:287]: Thank you for sharing it. It appears that the code path is not covered in our CI test. We will fix it.
